### ISSUE TITLE
`multipass start` (multipass 0.8.0+mac) timing out on MacOS 10.15 (Catalina)

### ISSUE THREAD

**Issue by squidsoup** [canonical/multipass#1122]
`multipass start bionic` (for a container that was working in MacOS 10.14) results in “timed out waiting for response”.  Running `multipass -v start bionic` doesn’t reveal anything further.

`multipass launch` does appear to work, and new containers can be created.

**Comment by Saviq** [comment]
Hi @squidsoup, thanks for reporting.

Can you have a look in `/Library/Logs/Multipass/`, there should be console logs of the machines as they start. Can you see them getting IP addresses? Can you SSH to them using the key in `/var/root/Library/Application Support/multipassd/ssh-keys/id_rsa`?

**Comment by cassiocassio** [comment]
I'm not disagreeing that you have issues, but your mileage may vary, as they say.
multipass  0.8.0+mac

 
<img width="922" alt="Screenshot 2019-10-11 at 09 21 50" src="https://user-images.githubusercontent.com/4593556/66636392-04267200-ec09-11e9-8587-9b9c23ba6e72.png">

**Comment by buggtb** [comment]
also.. as I have found, if you build electron apps on your laptop... unlucky ;)

**Comment by Saviq** [comment]
I can confirm this - instances start up fine and you can find their IP from:
```
cat /Library/Logs/Multipass/<instance_name>-hyperkit.log | tr -d '^G'
```

And then ssh into them:
```
ssh multipass@192.168.64.<last_ip_number> -i /var/root/Library/Application\ Support/multipassd/ssh-keys/id_rsa
```

I'm digging into what's going on.

**Comment by Saviq** [comment]
Oh and the above actually seems to make the instances work with `multipass start` again, too.

**Comment by Saviq** [comment]
**Actually** this is a race, folks, just try a couple `multipass start`/`multipass stop` cycles and the instances should come back. We're working on getting rid of the race.

**Comment by squidsoup** [comment]
Thanks @Saviq, assumings the logs aren't useful to you now?

**Comment by Saviq** [comment]
Yeah @squidsoup thanks, I can reproduce and have a plan to fix.

**Comment by Saviq** [comment]
Hi all, I've [a pre-release build](https://transfer.sh/UQoZK/multipass-0.9.0-dev.388+gfc97f60c.mac-Darwin.pkg) for you to test.

Since it's unsigned, you'll likely need to install with:
```
sudo installer -target / -verbose -pkg multipass-0.9.0-dev.388+gfc97f60c.mac-Darwin.pkg
```

**Comment by squidsoup** [comment]
@Saviq this appears to have solved the issue, thank you! 👏

**Comment by Saviq** [comment]
w00t!

**Comment by arashkaffamanesh** [comment]
the `multipass-0.9.0-dev.388+gd30fb679.mac-Darwin.pkg` doesn't work on Catalina, if you run multiple multipass VMs in parallel, e.g. like:

```
NODES=$(echo node{1..3})

# Create containers
for NODE in ${NODES}; do multipass launch --name ${NODE} --cpus 2 --mem 4G --disk 10G; done
```

it times out and in the logs I could see it can't get IPs.

**Comment by Saviq** [comment]
Hi @arashkaffamanesh 

> doesn't work on Catalina, if you run multiple multipass VMs in parallel, e.g. like:

Your example doesn't actually launch multiple nodes in parallel, but one after the other.

It also worked fine here:
```
$ for NODE in node{1..3}; do multipass launch --name ${NODE} --cpus 1 --mem 1G; done
Launched: node1                                                                 
Launched: node2                                                                 
Launched: node3                                                                 
$ multipass ls 
Name                    State             IPv4             Image
node1                   Running           192.168.64.6     Ubuntu 18.04 LTS
node2                   Running           192.168.64.7     Ubuntu 18.04 LTS
node3                   Running           192.168.64.8     Ubuntu 18.04 LTS
```

Does launching a single instance with your high resource requirements work?

**Comment by Saviq** [comment]
@arashkaffamanesh I also tried it in parallel for real, and worked too:
```
$ for NODE in node{1..3}; do multipass launch --name ${NODE} --cpus 1 --mem 1G &; done
[2] 16630
[3] 16631
[4] 16632
Launched: node2                                                                 
                                                                                
[3]  - 16631 done       multipass launch --name ${NODE} --cpus 1 --mem 1G
Launched: node1                                                                 
                                                                                
[2]  - 16630 done       multipass launch --name ${NODE} --cpus 1 --mem 1G
Launched: node3                                                                 
                                                                                
[4]  + 16632 done       multipass launch --name ${NODE} --cpus 1 --mem 1G
$ multipass ls
Name                    State             IPv4             Image
node1                   Running           192.168.64.6     Ubuntu 18.04 LTS
node2                   Running           192.168.64.7     Ubuntu 18.04 LTS
node3                   Running           192.168.64.8     Ubuntu 18.04 LTS
```

**Comment by arashkaffamanesh** [comment]
@Saviq thanks so much for your kind feedback!

> Does launching a single instance with your high resource requirements work?

Yes, even multi node worked with multipass v0.8 before Catalina very nice, but now with a single instance it doesn't work:

```
$ multipass launch --name node1 --cpus 2 --mem 4G --disk 10G
launch failed: The following errors occurred:
node1: timed out waiting for response
```

here is the multipassd.log:

https://pastebin.com/1nLQ006M

I guess the relevant part is here in the log:

```
[2019-10-31T12:10:52.914] [info] [node1] fcntl(F_PUNCHHOLE) failed: host filesystem does not support sparse files: Operation not permitted
```

And here are the repos which worked very nice for me and others 100x before the Catalina era :-)
https://github.com/arashkaffamanesh/multipass-rke-rancher
https://github.com/arashkaffamanesh/multipass-k3s-rancher

I saw there is a Release v0.9.0 PR:
https://github.com/CanonicalLtd/multipass/pull/1135

But the download doesn't work for darwin.

Thanks again!

**Comment by Saviq** [comment]
> I guess the relevant part is here in the log:
> 
> ```
> [2019-10-31T12:10:52.914] [info] [node1] fcntl(F_PUNCHHOLE) failed: host filesystem does not support sparse files: Operation not permitted
> ```

No, that's benign.

It's actually this:
> ```
> [2019-10-31T12:10:52.877] [info] [node1] setting up watch for PTY
> ```

That means you're running a version without the fix for this bug.

> I saw there is a Release v0.9.0 PR:
> #1135
> 
> But the download doesn't work for darwin.

It seems to work again now. (transfer.sh has been having troubles, we're looking to move to a different place to host those).

I've uploaded the latest RCs to https://github.com/CanonicalLtd/multipass/releases/tag/v0.9.0-rc, too. These should work for you.

**Comment by arashkaffamanesh** [comment]
THX!
With this release I'm getting:

https://pastebin.com/jumVWKA2

```
$ multipass launch --name node1 --cpus 1 --mem 1G
launch failed: The following errors occurred:
Instance stopped while starting
```

**Comment by rickw** [comment]
The pre-release of version 0.9.0 is no longer available. Any idea when it will release? or how to build on Mac?

**Comment by arashkaffamanesh** [comment]
@rickw (do it on your own risk!)
wget https://github.com/arashkaffamanesh/multipass-k3s-rancher/blob/master/multipass-0.9.0%2Bmac-Darwin.pkg
sudo installer -target / -verbose -pkg  multipass-0.9.0+mac-Darwin.pkg

**Comment by Saviq** [comment]
@rickw we repackaged it and it's back [on the release commit](https://github.com/CanonicalLtd/multipass/commit/474f2d551eb1a1288f0b8a81f0174c48a7b41528#commitcomment-35789190).

We'll release this week for sure (it may have to be unsigned, if Apple doesn't issue us signing keys in time…).

**Comment by srikrishna-bb** [comment]
facing the same issue, is this issue resolved, it works fine with virtualbox on mac , but fails with hyperkit

**Comment by planetf1** [comment]
I'm also seeing the same -- I had previously created an image, but a second one I tried to create is failing.
ie
```
multipass launch -vvv -c 2 -d 4G -m 4G -n microk8s
launch failed: The following errors occurred:
microk8s: timed out waiting for response
```
I am using hyperkit

**Comment by Saviq** [comment]
Hi @planetf1 can you please file a separate issue and provide a bit more detail? Is there a VPN in play maybe?

https://github.com/canonical/multipass/issues/new/choose

**Comment by planetf1** [comment]
> Hi @planetf1 can you please file a separate issue and provide a bit more detail? Is there a VPN in play maybe?
> 
> https://github.com/canonical/multipass/issues/new?labels=bug&template=bug_report.md

Thanks. will open up if it reoccurrs. worked after a reboot.

**Comment by 4pastor** [comment]
I just resolved the problem on my Mac M1 Big Sur by turning off FireVault and tried again. Now that I am up and running, how do I use the shell as a computer screen where I can surf the web, download apps, videos and etc.? Please help and thanks for you kind support!

What do I do from here to have a desktop?
